### ISSUE TITLE
concurrent.process calls constructor of object instead of __call__ on Macos

### ISSUE THREAD

**Issue by wvzytlyn** [noxdafox/pebble#88]
Python 3.9.8
OS: Macos 12.0.1, M1 chip
Pebble: 4.6.3

The following code:
```
from multiprocessing import freeze_support
from pebble import concurrent

class MyCallable:
	__name__ : str = "MyCallable"
	def __call__(self, x):
		return x

def run_it():

	my_target = concurrent.process(timeout=10)(MyCallable())
	print("my_target:", my_target)

	print("direct call:", MyCallable()(1))
	print("subprocess call:", my_target(1).result())


if __name__ == "__main__":
    freeze_support()
    run_it()
```

on macos gives the error:
`TypeError: MyCallable() takes no arguments` which is raised because the argument is passed to the constructor instead of being passed to the `__call__` method of the instance passed to `concurrent.process(timeout=10)`.

On `#23~20.04.1-Ubuntu SMP Mon Nov 15 14:03:19 UTC 2021` it gives the correct output:
```my_target: <function _process_wrapper.<locals>.wrapper at 0x7fa036413310>
direct call: 1
subprocess call: 1
```

**Comment by noxdafox** [comment]
It is generally a bad idea to execute objects methods in separate processes as the developer needs to be very aware of what are the implications. The object must be pickleable, any object related state will be split and exist into two separate copies which will be drifting away.

Is there a specific reason why you are trying to achieve the above? I can surely fix the root cause but the more I look at the above code and the more I am unsure it's a good idea.

**Comment by wvzytlyn** [comment]
Thanks for the response. From what you say, I understand that the following is a more desired use of concurrent.process function?

```from multiprocessing import freeze_support
from pebble import concurrent
from typing import Callable, Any

def call_any_callable(
		callable: Callable,
		arg: Any,
	) -> Any:
		return callable(arg)

class MyCallable:
	__name__ : str = "MyCallable"
	def __call__(self, x):
		return x

def run_it():

	my_target = concurrent.process(timeout=10)(call_any_callable)
	print("my_target:", my_target)

	print("direct call:", MyCallable()(1))
	print("subprocess call:", my_target(MyCallable(), 1).result())


if __name__ == "__main__":
    freeze_support()
    run_it()```

For me this solution is good enough, so if you believe this is safer regarding drifting state, perhaps an exception which explains the situation is enough. Thanks!

**Comment by noxdafox** [comment]
This is an opinion of mine and you'll surely find people disagreeing with it. It's not a limitation of `concurrent.futures` but rather a recommendation against. I believe mixing OOP and multiprocessing/multithreading is an anti-pattern. By doing so, you need to really know what you are doing and there are plenty of very tricky corner cases.

Let me show an example:
```python
import multiprocessing


class Foo:
    def __init__(self):
        self.bar = open('/home/noxdafox/hello.txt')

    def callme(self):
        print(self.bar.read())


def main():
    f = Foo()
    p = multiprocessing.Process(target=f.callme)
    p.start()
    p.join()


if __name__ == '__main__':
    main()
```

The above works with the `fork` [start method](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods) (Linux). It fails with the `spawn` [start method](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods) (Windows and recent MAC OS).

```
Traceback (most recent call last):
  File "/tmp/example.py", line 21, in <module>
    main()
  File "/tmp/example.py", line 15, in main
    p.start()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  ...
  File "/usr/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/usr/lib/python3.9/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle '_io.TextIOWrapper' object
```

The core issue is objects are intended to store state which can be either accessed directly or manipulated through the use of methods. Handling a state shared between processes or threads is a very tricky business and requires a deep understanding of the underlying internals including your OS capabilities.

Hence the recommendation to prefer functions. It's easier to encapsulate the logic correctly so it can be "shipped" and executed on separated threads or processes.

Anyways, I fixed the core issue of your error in 0f87f90 and I will include it in the next release.
